src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -400,4 +400,85 @@ describe('ReactServerRendering', function() {
       expect(markup.indexOf('hello, world') >= 0).toBe(true);
     });
   });
+
+  it('warns with a no-op when an async setState is triggered', function() {
+    var Component = React.createClass({
+      componentWillMount: function() {
+        this.setState({text: 'hello'});
+        setTimeout(() => {
+          this.setState({text: 'error'});
+        });
+      },
+      render: function() {
+        return <div onClick={() => {}}>{this.state.text}</div>;
+      },
+    });
+
+    spyOn(console, 'error');
```

Comment:
Historically we ran these tests in regular phantom in the browser, before jest really existed. We did that for a while before turning off last year. We wanted to maintain that option value moving forward and tying to closely to jest would make it harder (unless jest starts supporting that use case).